BundleClient/app/src/main/java/net/discdd/bundleclient/BundleClientWifiDirectService.java

```diff
@@ -216,8 +216,9 @@ private BundleExchangeCounts exchangeWith(WifiP2pDevice device) {
             return bundleTransmission.doExchangeWithTransport(device.deviceAddress, device.deviceName,
                                                               wifiDirectManager.getGroupOwnerAddress().getHostAddress(),
                                                               7777);
-        } catch (Exception e) {
+        } catch (Throwable e) {
             logger.log(WARNING, "Failed to connect to " + device.deviceName, e);
+            e.printStackTrace();
```

Comment:
I think the logger would print out the stack trace already. you can remove this line